src/components/Checkbox/Checkbox.js

```diff
@@ -45,15 +44,16 @@ const Checkbox = ({
         }}
         className={`${prefix}--checkbox`}
         id={id}
-        ref={
-          ref ||
-          (el => {
-            input = el;
-            if (input) {
-              input.indeterminate = indeterminate;
-            }
-          })
-        }
+        ref={el => {
+          if (el) {
+            el.indeterminate = indeterminate;
+          }
+          if (typeof ref === 'function') {
```

Comment:
The logic may be a bit hacky, an alternative here is turning `<Checkbox>` to stateful.

Reply:
look good, i prefer a little hacky-looking code, over making a stateful component if not otherwise necessary! it might be cool to see a little comment explaining the hack in these occurrences tho 😇 (if the hack is sensible enough to explain verbally at least)

Reply:
Thanks @lovemecomputer for your comment - Agreed with you that we need some explanation of this approach - @alisonjoseph?